vscode-dotnet-runtime-library/src/Acquisition/WinMacGlobalInstaller.ts

```diff
@@ -0,0 +1,319 @@
+
+        const validInstallerStatusCodes = ['0', '1641', '3010']; // Ok, Pending Reboot, + Reboot Starting Now
+        if(validInstallerStatusCodes.includes(installerResult))
+        {
+            return '0'; // These statuses are a success, we don't want to throw.
+        }
+        else
+        {
+            return installerResult;
+        }
+    }
+
+    /**
+     *
+     * @param installerUrl the url of the installer to download.
+     * @returns the path to the installer which was downloaded into a directory managed by us.
+     */
+    private async downloadInstaller(installerUrl : string) : Promise<string>
+    {
+        const ourInstallerDownloadFolder = IGlobalInstaller.getDownloadedInstallFilesFolder();
+        this.file.wipeDirectory(ourInstallerDownloadFolder);
+        const installerPath = path.join(ourInstallerDownloadFolder, `${installerUrl.split('/').slice(-1)}`);
+
+        const installerDir = path.dirname(installerPath);
+        if (!fs.existsSync(installerDir)){
+            fs.mkdirSync(installerDir);
+        }
+
+        await this.webWorker.downloadFile(installerUrl, installerPath);
+        try
+        {
+            fs.chmodSync(installerPath, 0o744);
+        }
+        catch(error : any)
+        {
+            this.acquisitionContext.eventStream.post(new SuppressedAcquisitionError(error, `Failed to chmod +x on ${installerPath}.`));
+        }
+        return installerPath;
+    }
+
+    private async installerFileHasValidIntegrity(installerFile : string) : Promise<boolean>
+    {
+        const realFileHash = await this.file.getFileHash(installerFile);
+        const expectedFileHash = this.installerHash;
+
+        if(expectedFileHash === null)
+        {
+            const yes = validationPromptConstants.allowOption
+            const no = validationPromptConstants.cancelOption;
+            const message = validationPromptConstants.noSignatureMessage;
+
+            const pick = await this.utilityContext.ui.getModalWarningResponse(message, no, yes);
+            const userConsentsToContinue = pick === yes;
+            return userConsentsToContinue;
+        }
+
+        if(realFileHash !== expectedFileHash)
+        {
+            return false;
+        }
+        else
+        {
+            return true;
+        }
+    }
+
+    public async getExpectedGlobalSDKPath(specificSDKVersionInstalled : string, installedArch : string) : Promise<string>
+    {
+        if(os.platform() === 'win32')
+        {
+            // The program files should always be set, but in the off chance they are wiped, we can try to use the default as backup.
+            // Both ia32 and x64 machines will use 'Program Files'
+            // We don't anticipate a user would need to install the x86 SDK, and we don't have any routes that support that yet.
+            return process.env.programfiles ? path.resolve(path.join(process.env.programfiles, 'dotnet', 'sdk')) : path.resolve(`C:\\Program Files\\dotnet\\sdk\\`);
```

Comment:
This assumes that Program Files reside on a specific drive. That's not necessarily the case. Also, what about x64 SDK on arm64?

Reply:
This is true. This path will not be used unless the environment variable programfiles is not set. Basically, it is a last ditch effort if someone deleted the environment variables on their machine somehow, which for some reason there exists people out there will do that...

This currently only installs the .NET SDK that matches the machine architecture, so for now we don't need to worry about the case where there's an x64 SDK on arm64. Is there a large contingent of users that need that for some reason? If so, it would be good to know. Would you suggest adding the x64 to the path here in that corner-case?

Reply:
Emulation was a big deal during .NET 6.0, but now that we have VS support for arm64, it's probably less of a concern for SDKs, and mostly just going to be runtimes where users get applications that were built for x64 and want to run those on arm64 machine.